Replicate API Added!

zsxkib opened the pull request:
Hey @nari-labs team! 👋

Dia is great! The way it generates dialogue with speaker tags and handles things like `(laughs)` is really well done.

To help people try Dia and use it in their projects, I've put it on Replicate:

**Demo & API:** [https://replicate.com/zsxkib/dia](https://replicate.com/zsxkib/dia)

*I also posted about it on X/Twitter here: [https://x.com/zsakib_/status/1915037657064759716](https://x.com/zsakib_/status/1915037657064759716)*

**What's new in this PR:**

This PR adds the necessary files and configuration to run the existing Dia model on Replicate:

*   Added `predict.py`: A script to load the model and run it using Cog's interface.
*   Added `cog.yaml`: Defines the Python environment, system packages, and GPU requirements for Replicate.
*   Added `.dockerignore`: Optimizes the Docker build by excluding unnecessary files.
*   Updated `.gitignore`: Includes Replicate/Cog specific files like `.cog/` and `/model_cache/`.
*   Updated `README.md`: Added a Replicate badge linking to the demo page.

This setup provides:

1.  **A web interface** for easy testing without any local setup.
2.  **An API endpoint** for developers to run Dia generation from their own code.

**Example API Usage:**

```python
import replicate

output = replicate.run(
    "zsxkib/dia:VERSION_HASH", # Replace VERSION_HASH with the actual model version hash
    input={
        "text": "[S1] Hello there! How are you? [S2] I'm doing great, thanks for asking! (laughs)",
        "audio_prompt": "https://replicate.delivery/pbxt/YourAudioFile.mp3", # Optional
        "max_audio_prompt_seconds": 10, # Optional, controls prompt length (default 10s)
        "temperature": 1.3,
        "cfg_scale": 3.0
        # ... other parameters ...
    }
)
print(output)
# Output: URL to the generated .wav file
```

On Replicate, users pay for their own compute time, which helps keep things running even if a model gets popular.

Let me know if you have any questions or feedback on the Replicate page. Happy to change anything.

Thanks for making Dia open source!

\- Sakib @ Replicate

buttercrab commented on README.md:
```diff
@@ -11,6 +11,7 @@
 <p align="center">
 <a href="https://huggingface.co/nari-labs/Dia-1.6B"><img src="https://huggingface.co/datasets/huggingface/badges/resolve/main/model-on-hf-lg-dark.svg" alt="Dataset on HuggingFace" height=42 ></a>
 <a href="https://huggingface.co/spaces/nari-labs/Dia-1.6B"><img src="https://huggingface.co/datasets/huggingface/badges/resolve/main/open-in-hf-spaces-lg-dark.svg" alt="Space on HuggingFace" height=38></a>
+<a href="https://replicate.com/zsxkib/dia"><img src="https://replicate.com/zsxkib/dia/badge" alt="Run on Replicate" height=38></a>
```
```suggestion
```

buttercrab reviewed (changes requested):
Can you move these changes under `/example/replicate`?

zsxkib commented:
Hi @buttercrab!

Thanks for the feedback! I've updated the PR based on your suggestion.

Specifically, I've:
*   Moved the Replicate-specific files (`predict.py` and `cog.yaml`) into a new `example/replicate/` directory.
*   Created a new `README.md` within `example/replicate/` which now includes the Replicate badge, API usage examples, and a brief explanation of the files.
*   Updated the main `README.md` to remove the badge and instead link to this new example directory and its README.

The Replicate badge is now only in the `example/replicate/README.md`. I wasn't sure if you'd prefer it to be completely removed or if having it within the example-specific README is okay. Please let me know if you'd like any further adjustments to the READMEs or anything else!

Thanks again for your guidance!

Best,
Sakib

buttercrab commented on README.md:
```diff
@@ -12,6 +12,9 @@
 <a href="https://huggingface.co/nari-labs/Dia-1.6B"><img src="https://huggingface.co/datasets/huggingface/badges/resolve/main/model-on-hf-lg-dark.svg" alt="Dataset on HuggingFace" height=42 ></a>
 <a href="https://huggingface.co/spaces/nari-labs/Dia-1.6B"><img src="https://huggingface.co/datasets/huggingface/badges/resolve/main/open-in-hf-spaces-lg-dark.svg" alt="Space on HuggingFace" height=38></a>
 </p>
+<p align="center">
```
Please remove this

zsxkib commented:
Hi @buttercrab,

I've removed the paragraph referencing the Replicate example from the main `README.md` as requested.

The commit for this is `cb50d4d`.

Please let me know if there's anything else!

Best,
Sakib
